File: Models/ViewModels/PostPromotionVM.cs

namespace GoWheels_WebAPI.Models.ViewModels
{
    public class PostPromotionVM
    {
        public int Id { get; set; }
        public int PromotionId { get; set; }
        public PromotionVM Promotion { get; set; } = null!;
        public int PostId { get; set; }
        public string? PostName { get; set; }
    }
}

File: FE/src/Model/PostPromotionVM.js
// src/models/PostPromotionVM.js

export default class PostPromotionVM {
    constructor(data = {}) {
      this.id = data.id || 0;
      this.promotionId = data.promotionId || 0;
      this.promotionContent = data.promotionContent || '';
      this.postId = data.postId || 0;
      this.postName = data.postName || '';
    }
  }
  